web3/src/main/java/org/hiero/mirror/web3/state/keyvalue/AbstractAliasedAccountReadableKVState.java

```diff
@@ -165,12 +165,20 @@ private Supplier<Long> getAccountBalance(final Entity entity, final Optional<Lon
     }
 
     private Long getBalanceOrDefaultToMinimum(final Entity entity, final Long balance) {
+        final ContractCallContext context = ContractCallContext.get();
+
+        boolean shouldOverride;
         if (!mirrorNodeEvmProperties.isOverridePayerBalanceValidation()) {
+            shouldOverride = false;
+        } else {
+            shouldOverride = context.shouldOverridePayerBalanceValidation();
+        }
+
+        if (!shouldOverride) {
             return Objects.requireNonNullElse(balance, 0L);
         }
 
-        final ContractCallContext context = ContractCallContext.get();
-        final var isBalanceCall = context.isBalanceCallSafe();
+        final var isBalanceCall = ContractCallContext.isBalanceCallSafe();
```

Comment:
Getting the context is relatively expensive.

```suggestion
        final var isBalanceCall = context.isBalanceCallSafe();
```